src/main/java/com/redhat/devtools/lsp4ij/features/documentSymbol/LSPDocumentSymbolBreadcrumbsInfoProvider.java

```diff
@@ -0,0 +1,152 @@
+/*******************************************************************************
+ * Copyright (c) 2025 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ *
+ * Contributors:
+ *     Red Hat Inc. - initial API and implementation
+ *******************************************************************************/
+
+package com.redhat.devtools.lsp4ij.features.documentSymbol;
+
+import com.intellij.ide.structureView.StructureViewTreeElement;
+import com.intellij.lang.Language;
+import com.intellij.lang.LanguageUtil;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Conditions;
+import com.intellij.openapi.vfs.VirtualFile;
+import com.intellij.psi.PsiElement;
+import com.intellij.psi.PsiFile;
+import com.intellij.ui.breadcrumbs.BreadcrumbsProvider;
+import com.intellij.util.containers.ContainerUtil;
+import com.redhat.devtools.lsp4ij.LanguageServiceAccessor;
+import com.redhat.devtools.lsp4ij.client.features.EditorBehaviorFeature;
+import com.redhat.devtools.lsp4ij.features.documentSymbol.LSPDocumentSymbolStructureViewModel.LSPDocumentSymbolViewElement;
+import com.redhat.devtools.lsp4ij.features.documentSymbol.LSPDocumentSymbolStructureViewModel.LSPFileStructureViewElement;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import javax.swing.Icon;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Document symbol-based breadcrumbs info provider.
+ */
+public class LSPDocumentSymbolBreadcrumbsInfoProvider implements BreadcrumbsProvider {
+
+    @Override
+    public Language[] getLanguages() {
+        // Register for all languages and filter in isSupported()
+        List<Language> languages = LanguageUtil.getLanguages(Conditions.alwaysTrue());
```

Comment:
Instead of registering for distinct languages in `plugin.xml`, this EP returns the supported languages. Since we don't know the full set at packaging time, it just returns all languages and then filtering occurs in the respective method implementations. This allows it the feature to work for all LSP4IJ files.

Reply:
LanguageServerregistry maintains a list of language where LS are associated in distinctLanguages. if you define a getter to this distinctLanguages, is it working for you?

Reply:
I'm having to reach back in my memory a few weeks, but the `getLanguages()` method is only called once for the `breadcrumbsInfoProvider`, so if language server definitions are configured at runtime, the IDE would have to be restarted to pick up those changes. This ensures that it's active for all languages, but I've registered it last in `plugin.xml` so that any more specific one for a given language would be used instead.